src/main/java/org/maxgamer/quickshop/shop/ContainerShop.java

```diff
@@ -401,19 +428,30 @@ public void delete(boolean memoryOnly) {
         if (memoryOnly) {
             // Delete it from memory
             plugin.getShopManager().removeShop(this);
+            onUnload();
         } else {
             // Delete the signs around it
             for (Sign s : this.getSigns()) {
                 s.getBlock().setType(Material.AIR);
             }
+            if (displayItem != null) {
+                displayItem.remove();
+            }
```

Comment:
`onUnload()` method will do it

Reply:
Also good catch. I had lots of doubled code from the beginning when I wanted to make sure that items were properly being removed and added.